lib/spree/shipment_notice.rb

```diff
@@ -10,25 +10,57 @@ def initialize(params)
     end
 
     def apply
-      locate ? update : not_found
+      find_shipment
+
+      unless shipment
+        log_not_found
+        return false
+      end
+
+      unless capture_payments!
+        log_not_paid
+        return false
+      end
+
+      ship_it
     rescue => e
```

Comment:
WOw! Awesome! I can now only read this piece of the code, and not even review anything below, as it is all private, and the tests ensure it is working.

I would've only added a bang to `ship_it!`, but even like that it's very very easy to read! Good stuff!